content/courses/Ada_For_The_C_Embedded_Developer/chapters/06_Translation.rst

```diff
@@ -725,9 +725,467 @@ above could also be literally translated to:
        Put_Line ("Value = " & Value_Type'Image (Value));
     end Main;
 
-Converting structures to Integer or Addresses
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Serialization of data types
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. todo::
+In the previous section, we've seen how to perform bitwise operations. In this
+section, we look at how to interpret a data type as a bit-field and perform
+low-level operations on it.
 
-    Complete section!
+In general, you can create a bit-field from any arbitrary data type. First, we
+declare a bit-field type like this:
+
+[Ada]
+
+.. code-block:: ada
+
+    type Bit_Field is array (Natural range <>) of Boolean with Pack;
+
+As we've seen previously, the :ada:`Pack` aspect declared at the end of the
+type declaration indicates that the compiler should optimize for size. We must
+use this attribute to be able to interpret data types as a bit-field.
+
+Then, we can use the :ada:`Size` and the :ada:`Address` attributes of an
+object of any type to declare a bit-field for this object. The :ada:`Size`
+attributes indicates the number of bits required to represent the object, while
+the :ada:`Address` attribute indicates the address in memory of that object.
+For example, assuming we've declare a variable :ada:`V`, we can declare an
+actual bit-field object using this pattern:
+
+[Ada]
+
+.. code-block:: ada
+
+    B : Bit_Field (0 .. V'Size - 1) with Address => V'Address;
+
+This technique is called overlays for serialization. Now, any operation that we
+perform on :ada:`B` will have a direct impact on :ada:`V`, since both are using
+the same memory location.
+
+Let's look at a simple example:
+
+[Ada]
+
+.. code:: ada run_button project=Courses.Ada_For_C_Embedded_Dev.Translation.Bitfield_Ada
+
+    with Ada.Text_IO; use Ada.Text_IO;
+
+    procedure Simple_Bitfield is
+       type Bit_Field is array (Natural range <>) of Boolean with Pack;
+
+       V : Integer := 0;
+       B : Bit_Field (0 .. V'Size - 1) with Address => V'Address;
+    begin
+       B (2) := True;
+       Put_Line ("V = " & Integer'Image (V));
```

Comment:
For the record, this is a case where volatile on V may be needed, in particular at high level of optimizations.